core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1473,7 +1473,20 @@ public Optional<TrinoPrincipal> getSchemaOwner(Session session, CatalogSchemaNam
     @Override
     public boolean isView(Session session, QualifiedObjectName viewName)
     {
-        return getViewInternal(session, viewName).isPresent();
+        if (cannotExist(viewName)) {
+            return false;
+        }
+
+        Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, viewName.catalogName());
+        if (catalog.isPresent()) {
+            CatalogMetadata catalogMetadata = catalog.get();
+            CatalogHandle catalogHandle = catalogMetadata.getCatalogHandle(session, viewName);
+            ConnectorMetadata metadata = catalogMetadata.getMetadataFor(session, catalogHandle);
+
+            ConnectorSession connectorSession = session.toConnectorSession(catalogHandle);
+            return metadata.isView(connectorSession, viewName.asSchemaTableName());
```

Comment:
This is the same as what's done in getViewInternal to resolve the right catalogHandle/ConnectorSession for plumbing through `isView`. It's a bit complex to abstract this behind common logic, and makes it less readable so I opted for just duplicating here.